000458/FlatironInstitute/000_lindi_vs_fsspec_streaming.ipynb

```diff
@@ -0,0 +1,146 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/magland/miniconda3/envs/000458_fi/lib/python3.12/site-packages/hdmf/utils.py:668: UserWarning: Ignoring cached namespace 'hdmf-common' version 1.5.1 because version 1.8.0 is already loaded.\n",
+      "  return func(args[0], **pargs)\n",
+      "/home/magland/miniconda3/envs/000458_fi/lib/python3.12/site-packages/hdmf/utils.py:668: UserWarning: Ignoring cached namespace 'core' version 2.5.0 because version 2.7.0 is already loaded.\n",
+      "  return func(args[0], **pargs)\n",
+      "/home/magland/miniconda3/envs/000458_fi/lib/python3.12/site-packages/hdmf/utils.py:668: UserWarning: Ignoring cached namespace 'hdmf-experimental' version 0.2.0 because version 0.5.0 is already loaded.\n",
+      "  return func(args[0], **pargs)\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Elapsed time for lindi: 7.52 s\n",
+      "Elapsed time for fsspec: 141.01 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "# This notebook compares the time taken to stream an NWB file from DANDI archive\n",
+    "# using lindi and fsspec. On my laptop on my home WiFi network, lindi streaming\n",
+    "# took 7.52 s and fsspec streaming took 141.01 s. This is expected to depend\n",
```

Comment:
any chance to add timing comparison to your nwb-tuned alternative to fsspec for a more frank comparison?  ;-)

Reply:
I would expect lindi to be about the same as remfile, but I'm moving toward using lindi for everything since it has benefits beyond just this streaming. The purpose here is not to do a rigorous comparison between the methods, it's just to motivate why I'm not using fsspec, which seems to be the standard recommended method in all the docs still. So I'd prefer not to also include remfile here.